src/trace/trace.cpp

```diff
@@ -656,6 +667,40 @@ otf2::definition::mapping_table Trace::merge_calling_contexts(ThreadCctxRefMap&
         otf2::definition::mapping_table::mapping_type_type::calling_context, mappings);
 }
 
+otf2::definition::mapping_table
+Trace::merge_syscall_contexts(const std::set<int64_t>& used_syscalls)
+{
+    // TODO: insert real number of syscalls here
+    std::vector<uint32_t> mappings(450);
+
+    for (const auto& syscall_nr : used_syscalls)
+    {
+        std::string syscall_str = fmt::format("syscall {}", syscall_nr);
+        auto syscall_entry = syscall_names().find(syscall_nr);
+        if (syscall_entry != syscall_names().end())
+        {
+            syscall_str = syscall_entry->second;
+        }
```

Comment:
This should be its own function, and then we wouldn't have to overwrite the result of format and could save a local variable. Also uncle bob would be proud.